packages/react-components/src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -191,21 +201,53 @@ export const RichTextEditor = React.forwardRef<RichTextEditorComponentRef, RichT
 
   useEffect(() => {
     // don't set callback in plugin constructor to update callback without plugin recreation
-    updatePlugin.onUpdate = (event: string, imageSrcArray?: Array<string>) => {
+    updatePlugin.onUpdate = (
+      event: string,
+      /* @conditional-compile-remove(rich-text-editor-image-upload) */ shouldUpdateInlineImages?: boolean
+    ) => {
       if (editor.current === null) {
         return;
       }
       if (event === UpdateEvent.Blur || event === UpdateEvent.Dispose) {
         onContentModelUpdate && onContentModelUpdate(editor.current.getContentModelCopy('disconnected'));
       } else {
-        onChange && onChange(exportContent(editor.current), imageSrcArray);
+        const content = exportContent(editor.current);
+        /* @conditional-compile-remove(rich-text-editor-image-upload) */
+        let addedInlineImages: Record<string, string>[] = [];
+        /* @conditional-compile-remove(rich-text-editor-image-upload) */
+        let removedInlineImages: Record<string, string>[] = [];
+        /* @conditional-compile-remove(rich-text-editor-image-upload) */
+        if (shouldUpdateInlineImages) {
+          /* @conditional-compile-remove(rich-text-editor-image-upload) */
+          // We cannot get the added images directly from the copyPastePlugin because the onUpdate is called before we can set an internal state.
+          addedInlineImages = getAddedInlineImages(content, previousInlineImages);
```

Comment:
we can probably move these calculations to UpdatePlugin so the update logic will be in 1 place instead of having it in UpdatePlugin and Editor itself

Reply:
previousInlineImages can be a propert for UpdatePlugin then

Reply:
also in this case we can separate where `removedInlineImages` are calculate. For the changes of this PR, I don't think that we need `addedInlineImages`. Please see previous comments for plugins files for more details